src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1066,9 +1070,24 @@ private Task<Stream> InternalGetRequestStream()
                 throw new InvalidOperationException(SR.net_reqsubmitted);
             }
 
-            _requestStream = new RequestStream();
+            // If we aren't buffering we need to open the connection right away.
+            // Because we need to send the data as soon as possible when it's available from the RequestStream.
+            // Making this allows us to keep the sync send request path for buffering cases.
+            if (AllowWriteStreamBuffering is false)
+            {
+                // We're calling SendRequest with async, because we need to open the connection and send the request
+                // Otherwise, sync path will block the current thread until the request is sent.
+                TaskCompletionSource<Stream> getStreamTcs = new();
+                TaskCompletionSource completeTcs = new();
+                _sendRequestTask = SendRequest(async: true, new RequestStreamContent(getStreamTcs, completeTcs));
+                _requestStream = new RequestStream(getStreamTcs.Task.GetAwaiter().GetResult(), completeTcs);
```

Comment:
Looks like I can mark this method as async and await this `getStreamTcs.Task` instead of calling `GetAwaiter().GetResult()`

Reply:
Please do.